### PR TITLE
Add option to pass include/lib dirs for building with PowerAI DDL

### PR DESCRIPTION
**Pull request by nvcastet**
IBM PowerAI DDL will be installed in Anaconda environments. The path will not always be '/opt/DL/ddl' anymore.
HOROVOD_DDL_HOME or HOROVOD_DDL_INCLUDE/HOROVOD_DDL_LIB can be used to
specify include/lib dirs.

### CODE REVIEW

**Comment by nvcastet** [comment]
@alsrgv

File: setup.py [hunk 1]
```diff
@@ -429,20 +429,47 @@ def get_nccl_vals(build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags):
...
+    ddl_include_dir = os.environ.get('HOROVOD_DDL_INCLUDE')
+    if ddl_include_dir:
+        ddl_include_dirs += [ddl_include_dir]
+
+    ddl_lib_dir = os.environ.get('HOROVOD_DDL_LIB')
+    if ddl_lib_dir:
+        ddl_lib_dirs += [ddl_lib_dir]
+
+    # Keep DDL legacy folders for backward compatibility
+    ddl_include_dirs += ['/opt/DL/ddl/include']
```
Comment (alsrgv) [inline setup.py]: Should these be only added if the list is empty?

Comment (nvcastet) [reply setup.py]: yes that makes sense.

**Comment by nvcastet** [comment]
@alsrgv Would it possible to cut a new release for that?

**Comment by alsrgv** [comment]
@nvcastet, we'll cut 0.16.1 when #858 is merged, should be this or early next week.

**Comment by nvcastet** [comment]
@alsrgv Perfect, thank you!

**Comment by denfromufa** [comment]
@nvcastet is this tutorial for ddl+horovod still valid for powerai 1.6? 

https://developer.ibm.com/linuxonpower/2018/08/24/distributed-deep-learning-horovod-powerai-ddl/

Why is horovod not included with powerai?

**Comment by nvcastet** [comment]
@denfromufa The blog post will get updated for 1.6 soon. Meanwhile, here are the new commands to install Horovod in your conda environment:
```
conda install  ddl tensorflow gxx_linux-ppc64le=7.3 cffi cudatoolkit-dev -y
HOROVOD_CUDA_HOME=$CONDA_PREFIX HOROVOD_GPU_ALLREDUCE=DDL pip install horovod
```

**Comment by denfromufa** [comment]
@nvcastet I'm getting these errors even after your instructions:

```
    powerpc64le-conda_cos7-linux-gnu-cc: error: unrecognized command line option '-stdlib=libc++'
    powerpc64le-conda_cos7-linux-gnu-cc: error: unrecognized command line option '-mf16c'
    powerpc64le-conda_cos7-linux-gnu-cc: error: unrecognized command line option '-mavx'; did you mean '-mads'?
```

I attached the full log.

[pip_horovod_install_castet.txt](https://github.com/horovod/horovod/files/2998375/pip_horovod_install_castet.txt)

**Comment by denfromufa** [comment]
BTW, just MPI or NCCL linking builds without errors for Horovod on PowerAI 1.6:

HOROVOD_GPU_ALLREDUCE=MPI pip install --no-cache-dir horovod

or 

HOROVOD_GPU_ALLREDUCE=NCCL pip install --no-cache-dir horovod

**Comment by alsrgv** [comment]
@denfromufa, looks like we introduced a typo as part of previous operations.cc refactoring.  I've submitted #946 to fix, but we don't have any machine to test it.  Could you try it by installing Horovod like this (based on @nvcastet instruction): `conda install  ddl tensorflow gxx_linux-ppc64le=7.3 cffi cudatoolkit-dev -y
HOROVOD_CUDA_HOME=$CONDA_PREFIX HOROVOD_GPU_ALLREDUCE=DDL pip install git+https://github.com/horovod/horovod@fix_ddl`

**Comment by denfromufa** [comment]
@alsrgv the fix is looking good! @nvcastet why don't you provide the powerci for horovod like for semi-official TF builds?

https://powerci.osuosl.org/

**Comment by nvcastet** [comment]
@denfromufa I am glad it worked. So far, it has been straightforward to install Horovod with `pip` on ppc64le but I will bring your suggestion to our team.

**Comment by denfromufa** [comment]
cc @wdirons with whom we had a discussion about horovod for powerai and powerci.

**Comment by denfromufa** [comment]
@nvcastet can i install ddl for more than 4 nodes with conda? currently it requires a whole different powerai enterprise installation.

**Comment by bnemanich** [comment]
@denfromufa the DDL that is available for free as part of PowerAI can work on up to 4 nodes. In order to use more than 4 nodes, you will need a license for Watson Machine Learning Accelerator (formally known as PowerAI Enterprise).

**Pull request merged**